exercises/practice/strain/strain-test.rkt

```diff
@@ -0,0 +1,72 @@
+#lang racket/base
+
+(require "strain.rkt")
+
+(module+ test
+  (require rackunit rackunit/text-ui)
+
+  (define suite
+    (test-suite
+     "strain tests"
+
+     (test-equal? "keep on empty list returns empty list"
+                  (keep '() (lambda (x) #t))
+                  '())
+
+     (test-equal? "keeps everything"
+                  (keep '(1 3 5) (lambda (x) #t))
+                  '(1 3 5))
+
+     (test-equal? "keeps nothing"
+                  (keep '(1 3 5) (lambda (x) #f))
+                  '())
+
+     (test-equal? "keeps first and last"
+                  (keep '(1 2 3) odd?)
+                  '(1 3))
+
+     (test-equal? "keeps neither first nor last"
+                  (keep '(1 2 3) even?)
+                  '(2))
+
+     (test-equal? "keeps strings"
+                  (keep '("apple" "zebra" "banana" "zombies" "cherimoya" "zealot")
+                        (lambda (x) (equal? (string-ref x 0) #\z)))
+                  '("zebra" "zombies" "zealot"))
+
+     (test-equal? "keeps lists"
+                  (keep (list '(1 2 3) '(5 5 5) '(5 1 2) '(2 1 2) '(1 5 2) '(2 2 1) '(1 2 5))
+                        (lambda (x) (if (member 5 x) #t #f)))
```

Comment:
We can simplify this by removing the if. Any value not false is treated as true so the predicate can return the result of `(member 5 x)`.

Reply:
In fact, let's define that  as `has-value-of-five?` or something similar.